### PR TITLE
CE-5879: adding fingerprinting to the credential list

### PR DESCRIPTION
**Pull request by carlosrodlop**
Selection_054

https://github.com/jenkinsci/credentials-plugin/blob/master/src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java#L1277

### CODE REVIEW

File: list-credential.groovy [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
+import com.cloudbees.plugins.credentials.CredentialsProvider
 
-Set<Credentials> allCredentials = new HashSet<Credentials>();
-
-def creds = com.cloudbees.plugins.credentials.CredentialsProvider.lookupCredentials(
-      com.cloudbees.plugins.credentials.Credentials.class
-);
-
-allCredentials.addAll(creds)
+Set<Credentials> allCredentials = new HashSet<Credentials>()
```
Comment (fbelzunc) [inline list-credential.groovy:12]:
```suggestion
Set<Credentials> allCredentials = new HashSet<Credentials>();
```
so don't have any change here

File: list-credential.groovy [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
 
 Jenkins.instance.getAllItems(com.cloudbees.hudson.plugins.folder.Folder.class).each{ f ->
  creds = com.cloudbees.plugins.credentials.CredentialsProvider.lookupCredentials(
       com.cloudbees.plugins.credentials.Credentials.class, f)
   allCredentials.addAll(creds)
-  
+
```
Comment (fbelzunc) [inline list-credential.groovy:18]: remove this extra line

File: list-credential.groovy [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
 }
 
 for (c in allCredentials) {
-   println(c.id + ": " + c.description)
-}
+  if (CredentialsProvider.FINGERPRINT_ENABLED) {
+    fp = CredentialsProvider.getFingerprintOf(c)
+    if (fp){
```
Comment (fbelzunc) [inline list-credential.groovy:24]:
```suggestion
    if (fp) {  
```

File: list-credential.groovy [hunk 4]
```diff
@@ -1,24 +1,31 @@
...
+
 }
 
 for (c in allCredentials) {
-   println(c.id + ": " + c.description)
-}
+  if (CredentialsProvider.FINGERPRINT_ENABLED) {
+    fp = CredentialsProvider.getFingerprintOf(c)
+    if (fp){  
+    	fp_str="Fingerprinted jobs: " + fp.getJobs()
```
Comment (fbelzunc) [inline list-credential.groovy:25]:
```suggestion
    	fp_str = "Fingerprinted jobs: " + fp.getJobs()
```

File: list-credential.groovy [hunk 5]
```diff
@@ -1,24 +1,31 @@
...
+  	} else {
+    	fp_str="(No Fingerprints)"
```
Comment (fbelzunc) [inline list-credential.groovy:27]:
```suggestion
    	fp_str = "(No Fingerprints)"
```

**Review by fbelzunc — CHANGES_REQUESTED** [review]

File: list-credential.groovy (hunk 1 above)
Comment (carlosrodlop) [reply list-credential.groovy:12]: @fbelzunc, From my understanding when it compiles without semicolons that means that they are not needed https://dzone.com/articles/groovy-sometimes-you-still WDTY?

**Review by fbelzunc — APPROVED** [review]

**Pull request merged**